imodels/tree/hierarchical_shrinkage.py

```diff
@@ -317,3 +362,15 @@ def fit(self, X, y, *args, **kwargs):
     # m.predict_proba(X_train)  # just run this
     # print('score', m.score(X_test, y_test))
     print('score', r2_score(y_test, m.predict(X_test)))
+
+    m = HSTreeRegressorCV(estimator_=DecisionTreeRegressor(max_leaf_nodes=10, random_state=1),
+                           shrinkage_scheme_='node_based',
+                           reg_param_list=[0.1, 1, 2, 5, 10, 25, 50, 100, 500])
```

Comment:
Since we train the original model prior to adding the reg_param, it would make sense to also score the model without any reg_param (vanilla model, no HSTree), and if it gets a better val score than any of the reg_param values, then that should be the model returned.

Reply:
Good idea, let's add `0` to `reg_param_list`.